### PR TITLE
refactor(Blocks): add `RealInputArray` and `RealOutputArray`

### PR DESCRIPTION
**Pull request by sathvikbhagavan**
This adds `RealInputArray` and `RealOutputArray` components
This also changes the default values of the states in the RealInput/Output/Array components to guesses.

This is useful for using symbolic vector for input/output size to be 1. Needed for using Neural Network blocks. Ref - https://github.com/SciML/ModelingToolkitNeuralNets.jl/issues/16

### CODE REVIEW

**Review by YingboMa — CHANGES_REQUESTED** [review]
@ChrisRackauckas and I believe that this is not correct in general. `RealInput` should be for just scalars and we should add a separate `RealInputArray` component instead. For backward compat, we should keep `nin`, etc in `RealInput` and add a warning when `nin>1`.

**Comment by sathvikbhagavan** [comment]
@YingboMa @ChrisRackauckas, I have added the components for `RealInputArray` and `RealOutputArray`.

File: src/Blocks/utils.jl [hunk 1]
```diff
@@ -42,18 +64,38 @@ Connector with one input signal of type Real.
...
+    u = collect(u)
+    ODESystem(Equation[], t, [u...], []; name = name)
+end
+@doc """
+    RealOutputArray(;name, nout, u_start)
+
+Connector with an array of output signals of type Real.
+
+# Parameters:
+- `nout=2`: Number of outputs.
```
Comment (ChrisRackauckas) [inline src/Blocks/utils.jl:92]: That default makes no sense. Why `2`?

File: src/Blocks/utils.jl [hunk 2]
```diff
@@ -14,19 +15,40 @@
...
 @doc """
-    RealInput(;name, nin, u_start)
+    RealInput(;name, u_start)
 
 Connector with one input signal of type Real.
 
 # Parameters:
-- `nin=1`: Number of inputs
-- `u_start=0`: Initial value for `u`
+- `u_start=0`: Initial value for `u`.
```
Comment (ChrisRackauckas) [inline src/Blocks/utils.jl:23]: Why does this have an initial value default?

Comment (sathvikbhagavan) [reply src/Blocks/utils.jl:23]: Shouldn't there be a default value for the variable?

```julia
julia> @named x = RealInput(; nin = 1, u_start = 1.0)
Model x with 0 (1) equations
Unknowns (1):
  u(t) [defaults to 1.0]: Inner variable in RealInput x
Parameters (0):
```

File: src/Blocks/utils.jl (hunk 1 above)
Comment (sathvikbhagavan) [reply src/Blocks/utils.jl:92]: As this was an array, I thought 2 element might be more common than 1 element for the default. I can change it to 1 if thats better.

File: src/Blocks/utils.jl (hunk 2 above)
Comment (ChrisRackauckas) [reply src/Blocks/utils.jl:23]: Won't that cause issues in initialization?

File: src/Blocks/utils.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/Blocks/utils.jl:92]: Why would it have a default at all? This seems like it should be a required argument

Comment (sathvikbhagavan) [reply src/Blocks/utils.jl]: Yes, that would be better - addressed in [6bb905a](https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/283/commits/6bb905a45acfbdbf2517372ee99697e3c6de236a)

File: src/Blocks/utils.jl (hunk 2 above)
Comment (sathvikbhagavan) [reply src/Blocks/utils.jl]: Addressed in [4cfc041](https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/283/commits/4cfc041d9f6d242c48e5a5df41758d89ce23eb25)

Comment (sathvikbhagavan) [reply src/Blocks/utils.jl]: Removing the default is erroring - https://github.com/SciML/ModelingToolkitStandardLibrary.jl/actions/runs/8706828508/job/23880355450?pr=283#step:6:577 even though there is an initial guess 🤔

Comment (sathvikbhagavan) [reply src/Blocks/utils.jl]: After talking with Aayush, we found guesses are not propagated in a hierarchical system. Hopefully https://github.com/SciML/ModelingToolkit.jl/pull/2654 should fix it

Comment (sathvikbhagavan) [reply src/Blocks/utils.jl]: One more bug from MTK - `linearize_function` doesn't use guesses - https://github.com/SciML/ModelingToolkit.jl/pull/2656 should fix it

Comment (ChrisRackauckas) [reply src/Blocks/utils.jl]: Linearization shouldn't necessarily use guesses. It should make sure the algebraic conditions are satisfied and that's it. That's a separate issue though, just update the test to be good and we can merge, the linearization failure is separate from this.

Comment (sathvikbhagavan) [reply src/Blocks/utils.jl]: Do you mean to mark it as broken/skip? done that in [9bc4b8b](https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/283/commits/9bc4b8bceef58149cd00932bba791b54d208ae0b)

Comment (ChrisRackauckas) [reply src/Blocks/utils.jl]: no, fix it by passing in state values

**Pull request merged**